egs2/edacc/asr1/local/data.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+SECONDS=0
+
+
+stage=1
+stop_stage=100
+train_set="dev_train"
+valid_set="dev_non_train"
+test_set="test"
+sub_test_set="test_sub"
+
+log "$0 $*"
+. utils/parse_options.sh
+
+. ./db.sh
+. ./path.sh
+. ./cmd.sh
+
+if [ $# -ne 0 ]; then
+    log "Error: No positional arguments are required."
+    exit 2
+fi
+
+
+if [ -z "${EDACC}" ]; then
+    log "Fill the value of 'EDACC' of db.sh"
+    exit 1
+fi
+
+partitions="${train_set} ${valid_set} ${test_set} ${sub_test_set}"
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    if [ ! -e "${EDACC}/edacc_v1.0/README.txt" ]; then
+        echo "stage 1: Please download data from https://datashare.ed.ac.uk/handle/10283/4836 and save to ${EDACC}"
+    else
+        log "stage 1: ${EDACC}/edacc_v1.0/README.txt is already existing. Skip data downloading"
+    fi
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    log "stage 2: Data preparation"
+
+    # deal with too large wav file in data folder
+    audio_path="${EDACC}/edacc_v1.0/data/EDACC-C30.wav"
+    output_dir="${EDACC}/edacc_v1.0/data/segmentation"
+    mkdir -p "$output_dir"
+
+    if [ -f "$audio_path" ]; then
+        # segment at 1883 second
+        ffmpeg -i "$audio_path" -ss 0 -t 1883 "$output_dir/EDACC-C30_P1.wav"
+        ffmpeg -i "$audio_path" -ss 1883 -c copy "$output_dir/EDACC-C30_P2.wav"
+
+        echo "Audio file successfully split into:"
+        echo " - $output_dir/EDACC-C30_P1.wav"
+        echo " - $output_dir/EDACC-C30_P2.wav"
+    else
+        echo "File $audio_path not found. Please check the file path."
+        exit 1
+    fi
+
+    # prepare the date in Kaldi style, output will be "dev" folder and "test" folder in "data" folder
+    python3 local/data_prep.py "${EDACC}/edacc_v1.0" "data" "${output_dir}"
+
+    # # split the too long test utterance used for decoding section if necessary,
+    # # the alignment is based on CTC segmentation tool
+    # python3 local/truncate_test.py "data/test"
+
+    # # make training data from dev, as original data has no training data
+    # utils/subset_data_dir.sh --first data/dev 5000 "data/${train_set}"
+    # n=$(($(wc -l < data/dev/segments) - 5000))
+    # utils/subset_data_dir.sh --last data/dev ${n} "data/${valid_set}"
```

Comment:
As you mentioned in the PR, could you please explain a bit more on how you process the ctc segmentation? As I get into truncate test, I did not see any usage of CTC segmentation. It is still a bit unclear to me on how you fix the problem with that.

Additionally, according to your code, it seems that only EDACC-C30 is split?

Reply:
Since the process seems complicated, how about spliting the second stage into multiple smaller ones? We introduce stages, mostly for this purpose.

Reply:
> As you mentioned in the PR, could you please explain a bit more on how you process the ctc segmentation? As I get into truncate test, I did not see any usage of CTC segmentation. It is still a bit unclear to me on how you fix the problem with that.
> 
> Additionally, according to your code, it seems that only EDACC-C30 is split?

1. Yes, only EDACC-C30 is split, this is especially used for RM-shared in default memory case, to avoid OOM.
2. In truncate test, you should customize your segmentation rules, such as what text you need, start time and end time. And these parameters should be obtained in advance from a segmentation tool, I tested using segmentaion tool in my side, I didn't include actual text here because it will cause line too long error

Reply:
> Since the process seems complicated, how about spliting the second stage into multiple smaller ones? We introduce stages, mostly for this purpose.

Yes, I will do that

Reply:
I see, in that case, it would be helpful if you could make it clear in the script that this step is optional.

For a better understanding and future usage of the script you provided, you might consider putting an example in the readme with markdown style (won't trigger the length warning).

Reply:
Good advice, will do